packages/plugins/users-permissions/server/controllers/auth.js

```diff
@@ -347,12 +364,24 @@ module.exports = {
 
     await validateRegisterBody(params);
 
-    const role = await strapi
-      .query('plugin::users-permissions.role')
-      .findOne({ where: { type: settings.default_role } });
+     // Determine role to assign
+    let role;
+    if (ctx.request.body.role) {
+      role = await strapi.query('plugin::users-permissions.role').findOne({
+        where: { id: ctx.request.body.role },
+      });
+
+      if (!role) {
```

Comment:
This would be a breaking change

Reply:
Then what is the way to create user based on different role from API ?